typing/typeclass.ml

```diff
@@ -1976,6 +1995,10 @@ let report_error env ppf = function
   | Duplicate (kind, name) ->
       fprintf ppf "@[The %s `%s'@ has multiple definitions in this object@]"
                     kind name
+  | Closing_self_type self ->
+    fprintf ppf "@[Cannot close type of object literal:@ %a@.\
+                   Self types cannot be closed.@]"
+      Printtyp.type_scheme self
```

Comment:
I don't think people will understand this message...

Reply:
I think you could share the explanation that you added in the code itself, something like
```OCaml
@[Cannot close type of object literal:@ %a@,\
      Some self types have been unified with the self type of a not yet finished class,@ \
      they cannot be closed@]
```
(also using "@." should be avoided since it sends the formatting engine of Format in panic mode)